gpg/tasks/main.yml

```diff
@@ -6,12 +6,64 @@
     update_cache: yes
     cache_valid_time: 3600
 
+- name: Ensure {{ gpg_key_basedir }} config directory exists with right permissions
+  file:
+    dest: "~/{{ gpg_key_basedir }}"
+    state: directory
+    mode: 0700
+
+- name: Write private keys to {{ gpg_key_basedir }}
+  copy:
+    content: "{{ item.key|b64decode }}"
+    dest: "~/{{ gpg_key_basedir }}/{{ item.name }}"
+  with_items: "{{ gpg_private_keys }}"
+
+- name: Write private key passphrases to {{ gpg_key_basedir }}
+  copy:
+    content: "{{ item.passphrase }}"
+    dest: "~/{{ gpg_key_basedir }}/{{ item.name }}.passphrase"
+  with_items: "{{ gpg_private_keys }}"
+
+- name: Set gpg pinentry mode to loopback in config file.
```

Comment:
This can be set on the command line during key import. I think it would be better to do this rather than editing the user config.